plugins/kn-admin/core/root.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package core
+
+import (
+	"fmt"
+	"github.com/knative/client-contrib/plugins/kn-admin/cmd"
+	"github.com/knative/client-contrib/plugins/kn-admin/cmd/domain"
+	private_registry "github.com/knative/client-contrib/plugins/kn-admin/cmd/private-registry"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+
+func NewAdminCommand(params ...cmd.AdminParams) *cobra.Command {
+	p := &cmd.AdminParams{}
+	kubeConfig := os.Getenv("KUBECONFIG")
+	if kubeConfig == "" {
+		fmt.Println("cannot get cluster kube config, please export environment variable KUBECONFIG")
+		os.Exit(1)
+	}
+
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		fmt.Println("failed to build config:", err)
+		os.Exit(1)
+	}
+
+	p.ClientSet, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		fmt.Println("failed to create client:", err)
+		os.Exit(1)
+	}
+
+	rootCmd := &cobra.Command{
+		Use:   "admin",
+		Short: "A plugin of kn client to manage Knative",
+		Long: `A plugin of kn client to manage Knative for administrators. 
+
+For example:
+kn admin domain set - to set Knative route domain
+kn admin private-registry enable - to enable deployment from the private registry
+`,
+	}
+	cobra.OnInitialize(initConfig)
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.kn-admin.yaml)")
```

Comment:
Note command above with about new config fashion

Reply:
Also pay attention to how to deal with missing config file and how defaults are added. See kn version

Reply:
Updated with new config fashion.